### ISSUE TITLE
vm support: `Virtual Machine Installation` tutorial is misleading about the use case where VM auto-registration is not used

### ISSUE THREAD

**Issue by yskopets** [istio/istio#34579]
**Describe the feature request**

At the moment, [Virtual Machine Installation](https://istio.io/latest/docs/setup/install/virtual-machine/) tutorial includes 2 paths:
1) path that uses VM auto-registration
2) path that does not use VM auto-registration

The tutorial makes it seem these 2 paths produce equivalent results.

However, it's not the case.

On the path where VM auto-registration is not used:
* neither `WorkloadGroup` nor `WorkloadEntry` is created
  *  technically, `WorkloadGroup` is saved into a file, but it never applied to k8s
  * `WorkloadEntry` is not created neither explicitly nor implicitly
* yet, Istio Sidecar on the VM is allowed to connect to "istiod" and receive configuration (even though no such workload is described in the mesh either in the form of `WorkloadGroup` or `WorkloadEntry`)
* in the end, the VM onboarded such a way is not selectable neither by k8s `Service` objects nor by Istio `ServiceEntry` objects
  * in other words, the VM does not become a part of any load-balancing set, it will not receive any incoming requests

Expected:
* Tutorial should be clear about the difference between 2 paths
* Path that does not use VM auto-registration should provide instructions what extra configuration is necessary for VM to start receiving incoming requests

**Describe alternatives you've considered**

N/A

[X] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Additional context**

N/A

**Comment by cfanbo** [comment]
#34729
you can execute below command
$ kubectl get ws -A
$ kubectl get wg -A

WorkloadEntry as pod when instio install on vm. They can be selected！
